services/download/src/main/java/app/coronawarn/server/services/download/FederationBatchProcessor.java

```diff
@@ -42,6 +43,7 @@
   private final FederationGatewayDownloadService federationGatewayDownloadService;
   private final DownloadServiceConfig config;
   private final ValidFederationKeyFilter validFederationKeyFilter;
+  private final FifoMaxEntriesSet<FederationBatchInfo> cachedBatches;
```

Comment:
Did you check if there is a simpler solution for that? What about a list that stores all the processed batch tags and compare incoming "next batch tags" against this list?

Reply:
Imho we need to limit the amount of batches we lookup, or else we're prone to a memory leak. Therefore a simple list doesn't suffice. But again—very open to different solution proposals :-)

Reply:
Will just use a HashSet.

Reply:
Nice, a set makes indeed more sense than a list in this case :)